apps/snitch_core/lib/core/data/schema/payment/payment.ex

```diff
@@ -75,17 +75,4 @@ defmodule Snitch.Data.Schema.Payment do
   end
 
   defp validate_discriminator(changeset, _, _), do: changeset
```

Comment:
Do we need these arguments `validate_discriminator(:payment_type, @payment_types)`
As this function only validate payment types and cannot be reused as it specific for one thing.
This arguments can be removed and use them directly in the function definition iteself.
**OR** 
Ecto has similar thing for it:
`validate_inclusion(changeset, :payment_type, ["chk", "ccd"])`

Reply:
`validate_inclusion` only validates the `changes` but not the data of the changeset. All changeset functions assume that the `data` is always valid.
If I do the following, `validate_inclusion` will not complain whereas `validate_discriminator` will (use of `fetch_field`)
```elixir
Schema.Payment.create_changeset(
    %Schema.Payment{payment_type: "foo"},
    %{
      amount: Money.zero(:USD),
      payment_method_id: 2,
      slug: "foo",
      order_id: 3
    }
)
```

Generally, ecto validators work on `changes`, and I went paranoid in this case and tried to protect this extremely unlikely case.
```